src/commonMain/kotlin/app/cash/turbine/Turbine.kt

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (C) 2022 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.turbine
+
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.cancelAndJoin
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.Channel.Factory.UNLIMITED
+import kotlinx.coroutines.channels.ChannelResult
+
+internal const val debug = false
+
+/**
+ * A standalone [Turbine] suitable for usage in fakes or other external test components.
+ */
+public interface Turbine<T>: ReceiveTurbine<T>  {
+  /**
+   * Returns the underlying [Channel].
+   */
+  public override fun asChannel(): Channel<T>
+
+  public suspend fun cancel(cause: Throwable?)
+
+  /**
+   * Add an item to the underlying [Channel] without blocking.
+   *
+   * This method is equivalent to:
+   *
+   * ```
+   * if (!asChannel().trySend(item).isSuccess) error()
+   * ```
+   */
+  public fun add(item: T)
+
+  /**
+   * Assert that the next event received was non-null and return it.
+   * This function will not suspend, and will throw if invoked in a suspending context.
+   *
+   * @throws AssertionError if the next event was completion or an error.
+   */
+  public fun takeEvent(): Event<T>
+
+  /**
+   * Assert that the next event received was an item and return it.
+   * This function will not suspend, and will throw if invoked in a suspending context.
+   *
+   * @throws AssertionError if the next event was completion or an error, or no event.
+   */
+  public fun takeItem(): T
+
+  /**
+   * Assert that the next event received is [Event.Complete].
+   * This function will not suspend, and will throw if invoked in a suspending context.
+   *
+   * @throws AssertionError if the next event was completion or an error.
+   */
+  public fun takeComplete()
+
+  /**
+   * Assert that the next event received is [Event.Error], and return the error.
+   * This function will not suspend, and will throw if invoked in a suspending context.
+   *
+   * @throws AssertionError if the next event was completion or an error.
+   */
+  public fun takeError(): Throwable
+}
+
+public operator fun <T> Turbine<T>.plusAssign(value: T) { add(value) }
+
+/**
+ * Construct a standalone [Turbine] with an [UNLIMITED] buffer size.
```

Comment:
```suggestion
 * Construct a standalone [Turbine].
```
Since the channel is more an implementation detail now I think this would be a little confusing to include. If you want we can keep it on `asChannel()` to describe the underlying channel type.